### ISSUE TITLE
Option to toggle nearest neighbor filtering & anti-aliasing

### ISSUE THREAD

**Issue by ZechariahB** [maoschanz/drawing#565]
## Requested feature
Nearest neighbor is only enabled after an image has been zoomed >400% which leads to blurry pixel art for zoom less than 400%. Anti-aliasing is always enabled for transformations, which makes resizing and rotating less useful for pixel art. It would be great to toggle both of these as optional settings.

## Use case
Blurring at large enough zoom can make pixel art appear difficult to look at. Anti-aliasing can ruin color palettes by introducing new unwanted colors and ruins small sprites. Added option will eliminate these worries.

Fellow comrade. It also gives an excuse to hide more crustier dank hard boiled memes within Help.

**Comment by maoschanz** [comment]
>Nearest neighbor is only enabled after an image has been zoomed >400% which leads to blurry pixel art for zoom less than 400%

yes i didn't want to show the pixels for people not doing pixel art, and i assumed most sprites and artworks were small enough to require a zoom >400%

but if pixel art is your main use case i see why it's bothering you in general, it can indeed be an option. Meanwhile, you could see it as a feature: once your art is done, go to 200% to see how it would have been rendered on an old blurry CRT display! 😆

>Anti-aliasing is always enabled for transformations, which makes resizing and rotating less useful for pixel art

in theory, you should get no antialiasing at all in the following transformations:

- if you crop or flip in general
- if you scale up/down **by an integer factor**
- if you rotate **by an angle of 90/180/270°**

it's impossible to rotate precisely with the mouse when the zoom level is extreme _(this is caused by the way the pointer coordinates are always rounded up to the closest integer in order to, ironically, allow pixel art editing 😓)_, so you have to use the keyboard inputs to get 90/180/270. Only 3 possible angles is restricting, but tbh i've no idea how to rotate "correctly" pixel art by an angle of 31° 😅

i guess the main issue you're noticing is about the "integer factor" idea, because it depends on each pixel relative position, so in practice you get a 50% change of fucking up your art. For example this 24px image 
![image](https://user-images.githubusercontent.com/19996113/208387761-026b5477-479c-4770-a2a5-3927e4401322.png)
 scales down to this 12px image
![image](https://user-images.githubusercontent.com/19996113/208387813-0a47da9c-6c12-425b-b5be-951718fd9ca0.png)
which is half correct, but half wrong: the pixels are merged according to groups of 4, and if the groups are not those our eyes expect, it's disappointing (but not related to antialiasing tho)

aligning the pixels on odd/even row/columns to be sure they're on a position where they could be resized well is probably a pain to do manually (pressing ctrl to see the coordinates and mentally computing?) and i'd like to improve this with a grid overlay (as suggested by #437)

**Comment by Rudxain** [comment]
Rotation by skewing/shearing is an easy solution to this problem. See:
- [Matt Parker video](https://youtu.be/1LCEiVDHJmc)
- [A post](https://web.archive.org/web/20250106193355/https://cohost.org/tomforsyth/post/891823-rotation-with-three)
- [Interactive thing](https://github.andrewt.net/shears/)
- [Implementation details](https://www.ocf.berkeley.edu/~fricke/projects/israel/paeth/rotation_by_shearing.html)
